### ISSUE TITLE
lerna bootstrap --scope does not generate package-lock.json

### ISSUE THREAD

**Issue by naseemkullah** [lerna/lerna#2784]
tl;dr please confirm that `lerna bootstrap` (without further settings) generates/updates a package-lock file* but with `--scope` arg it does not, and wether this is intended or a bug.

*update: only true if `node_modules` does not exist

## Expected Behavior
When running `lerna bootstrap` without the `--scope` arg, `package-lock.json` is created.
I'd expect the same behaviour when using `--scope`.

## Current Behavior
`lerna bootstrap --include-dependencies --scope package-foo` does not generate a `package-lock.json`

## Possible Solution
n/a

## Steps to Reproduce (for bugs)

I'm including details about my setup that I don't think are totally related to the issue, but including them nonetheless.

1. create two lerna-managed folders, namely `services` (private npm packages, never published, consume shared code from other dir) and `packages` (dir that containers shared code packages).
2. Have a service package in `services` depend on a package from `packages`
3. run `lerna bootstrap --include-dependencies --scope <service name>` and observe no `package-lock.json` be generated
4. run `lerna bootstrap` and observer `package-lock.json` be generated

<details><summary>lerna.json</summary><p>

```json
{
  "packages": [
    "packages/*",
    "services/*"
  ],
  "version": "independent"
}
```
</p></details>

<details><summary>lerna-debug.log</summary><p>

```txt
<!-- If you have a `lerna-debug.log` available, please paste it here -->
<!-- Otherwise, feel free to delete this <details> block -->
```
</p></details>

## Context

We are trying to maintain a monorepo of containerized services (`services` dir) and their shared libraries (`packages` dir). We commit package-lock.json for all modules. If we accidentally run `npm install` the `package-lock.json` includes the sym-linked and unpublished shared lib, which has negative effect. We are pleased with the `package-lock.json` generated by `bootstrap` but want that same result when using `--scope`

## Your Environment

| Executable | Version |
all latest

| OS | Version |
| --- | --- |
| macOS Sierra | 10.12.3 |

**Comment by naseemkullah** [comment]
additional observed behaviour:

`lerna bootstrap` will only generate `package-lock.json` for packages who depend on other packages within monorepo.

`lerna boostrap --scope --include-dependencies` never does as per initial comment.

**Comment by naseemkullah** [comment]
Lerna you served us well, all the best

**Comment by DominicFarrington** [comment]
Hi this is still an issue, would love to see it resolved if possible.

**Comment by DominicFarrington** [comment]
Hi this is still an issue, would love to see it resolved if possible. @JamesHenry
